en/companion_computer/holybro_pixhawk_rpi_cm4_baseboard.md

```diff
@@ -205,13 +205,33 @@ On the RPi side:
 
 ## Ethernet Connection (Optional)
 
-The flight controller module is [internally connected to RPi CM4](#rpi-cm4-fc-serial-connection) from `TELEM2` (Serial).
+The flight controller module is [internally connected to RPi CM4](#rpi-cm4-fc-serial-connection) from `TELEM2` (serial).
 
 You can also set up a local Ethernet connection between them using the supplied cable.
 Ethernet connectivity provides a fast, reliable, and flexible communication alternative to using USB or other serial connections.
 
 ::: info
-For more general information see: [PX4 Ethernet Setup](../advanced_config/ethernet_setup.md).
+For general Ethernet setup information see: [PX4 Ethernet Setup](../advanced_config/ethernet_setup.md).
+
+The setup here is much the same, except that we have used the following `netplan` config on PX4:
+
+```sh
+network:
+  version: 2
+  renderer: NetworkManager
+  ethernets:
+    eth0:
+      addresses:
+        - 10.41.10.1/24
+      nameservers:
+        addresses: [10.41.10.1]
+      routes:
+        - to: 10.41.10.0/24  # Local route to access devices on this subnet
+          scope: link        # Scope link to restrict it to local subnet
+```
+
+This sets `eth0` as our channel for the local Ethernet link from the RPi (instead of `enp2s0`, which is assumed in [Ethernet Setup](../advanced_config/ethernet_setup.md#ubuntu-ethernet-network-setup)).
+Note that we only set up this locally linked route to prevent internet access disruption over WiFi.
```

Comment:
> Is this what you meant?
> 
> What does the second sentence mean? Perhaps you are suggesting that it is some kind of backup route?

We do not want the default route to be over local ethernet network cause then it blocks internet access. So we need to keep the default route over WiFi to have internet access. This is what I came up to with testing.

Reply:
@farhangnaderi Assume I know nothing. What is the/a default route? Do you mean the default route set on RPi or something else? What makes something the default route? Is it the name `eth0` or `enp2s0`?

What I "guess" you are saying is that we don't want to send Internet traffic over our local ethernet network because we're already using it for XRCE-DDS. If we were to send our internet packets the whole thing would fall over. But what are these internet packets you are sending? Connection to QGC over Wifi?

Anyway, so I think you're setting up a local link to send our XRCE-DDS packets over ETH, so we can keep our Wifi link open to QGC. Is that right? Presumably by setting up `eth0` we are telling the CM4 to use that for XRCE-DDS somehow?

If not, please spell it out in laymans networking.

Reply:
OK, so I'm just assuming I'm right - you could use Wifi for the comms,  but that would clash with internet. So we set up the local route.